unstructured_ingest/v2/utils.py

```diff
@@ -0,0 +1,45 @@
+import json
+from inspect import isclass
+from typing import Any
+
+from pydantic import BaseModel, field_serializer
+from pydantic.types import _SecretBase
+
+
+def is_secret(value: Any) -> bool:
+    # Case Secret[int]
+    if hasattr(value, "__origin__") and hasattr(value, "__args__"):
+        origin = value.__origin__
+        return isclass(value) and issubclass(origin, _SecretBase)
+    # Case SecretStr
+    return isclass(value) and issubclass(value, _SecretBase)
+
+
+def serialize_base_model(model: BaseModel) -> dict:
+    # To get the full serialized dict regardless of if values are marked as Secret
+    model_dict = model.dict()
+    for k, v in model_dict.items():
+        if isinstance(v, _SecretBase):
+            secret_value = v.get_secret_value()
```

Comment:
Can you write a test for this. I would sleep better knowing that we don't ever mess this up.

Reply:
Added